corehq/apps/reports/views.py

```diff
@@ -2032,14 +2033,14 @@ def _is_location_safe_report_class(view_fn, request, domain, export_hash, format
 @require_GET
 def export_report(request, domain, export_hash, format):
     cache = get_redis_client()
-
-    content = cache.get(export_hash)
-    if content is not None:
-        report_class, report_file = content
+    report_class = cache.get(export_hash)
```

Comment:
would it be possible to store `report_class` in some metadata for the blob so that we can remove all redis code from this path?

Reply:
I tried, but it looks like I can't just store a string in the blobdb? How do I store stuff in the metadata for blobdb?

Reply:
I think that's what `properties` is for https://github.com/dimagi/commcare-hq/blob/b3b4944ba8b92a55ad4ca5b4a88aed6de026615d/corehq/blobs/models.py#L63

Reply:
nifty

Reply:
You'll need the [`parent_id` and `key`](https://github.com/dimagi/commcare-hq/blob/3fcd83df8000666eb2ee0fc5b65f6b0718219508/corehq/blobs/metadata.py#L123) to retrieve metadata for the blob (which contains the `properties`). You can do something like this:

```py
try:
    meta = db.metadb.get(parent_id=export_hash, key=export_hash)
    report_class = meta.properties["report_class"]
except (corehq.blobs.models.BlobMeta.DoesNotExist, KeyError):
    return HttpResponseNotFound(...)

try:
    report_file = meta.open()
except corehq.blobs.NotFound:
    return HttpResponseNotFound(...)
with report_file:  # so it gets closed at the end
    # do stuff with report_file

Reply:
Update: improved error handling in code snippet above: no need to catch `KeyError` since that should practically never happen and could mask a fat-finger bug.

Reply:
https://github.com/dimagi/commcare-hq/pull/23927/commits/d20a40b6547fbcbaa58b30f4bb6939a89f3839af